prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -188,19 +190,19 @@ abstract class ConfigSchemaElement(
         when {
             !schema.isNullOrBlank() && value.isNotEmpty() ->
                 addError("Schema property cannot be used with the value property")
-            !schema.isNullOrBlank() && valueSet.isNotEmpty() ->
+            !schema.isNullOrBlank() && valueSet.getMapValues().isNotEmpty() ->
```

Comment:
Might be good to adjust the default to `null` or something to simplify some of these checks. We shouldn't need to build the map to be able to determine whether or not the schema has a `valueSet`.

Reply:
`valueSet` itself is not defined as a nullable type in `ConfigSchemaElement` so `null` wouldn't be a good check, but also the default value of `valueSet` for a schema that didn't provide one is an empty `InlineValueSet`, in which case `getMapValues` is working with the SortedMap directly. The only case a `LookupTableValueSet` would be empty is if the lookup table itself is empty or doesn't contain any values for the key and value columns, and we do need to build the map in order to detect that - though the case could be made that this should throw an exception instead.

Reply:
I don't mind making it a nullable type at this point. I would say an empty lookup table is a valid case, so if we're trying to enforce that `schema` and `valueSet` can't be specified in the same element, this misses that edge case.

Reply:
I don't think we have to worry about that edge case, since `valueSet` requires `value` and there is a check to prevent `schema` and `value` from being used together. I'm not necessarily against making it nullable, but I don't think it's necessary.

Reply:
Thats true, but the check for `valueSet` requiring `value` has the same problem. Take this case:
```
  - name: test-transform
    resource: 'Bundle.entry.resource.ofType(Patient).name'
    condition: 'Bundle.entry.resource.ofType(Patient).exists()'
    bundleProperty: '%resource.family'
    schema: someSchema.txt
    valueSet:
      lookupTable:
        tableName: some_empty_table
        keyColumn: first_name
        valueColumn: last_name
```
That schema is invalid for two reasons, 1) `schema` and `valueSet` are both present. 2) `valueSet` is present but `value` is not. Yet I think that case would be treated as valid because it doesn't look like we have a `valueSet` as far as these checks are concerned.
So we just need some way of knowing whether or not a `valueSet` is present, and just checking the resulting map isn't enough. Defaulting to null is one way, or you could add another function to the interface for `isConfigured` or something.

Reply:
I think setting a default value of null (as is the case for the majority of `ConfigSchema`) is the appropriate solution for this problem. I spent a little time on this but it's not a trivial effort to refactor the existing code and tests to handle this; I may spin this off into a separate issue since this isn't really in scope for this issue.

Reply:
I would say it is in scope mostly because it's an issue that is being introduced by these changes. And it's not just in this validation code, it's also in the element `merge` function, a valueSet won't be overwritten if the overwriting set it empty. I don't think it should be too complicated, should just be changing some default parameter values and then going through some of the valueSet logic and determining when `isEmpty` is appropriate and when `== null` is appropriate, but I wouldn't have expected any big refactor. If there's something I'm missing, let me know, maybe we can do some pairing to get through it.

Reply:
I took a stab at it today after my brief look at it last night. Most of the lift was in updating tests, and I happened to notice the specific tests weren't testing `valueSet` when they should have been.